src/sinks/influxdb/logs.rs

```diff
@@ -0,0 +1,657 @@
+use lazy_static::lazy_static;
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, HashMap};
+
+#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+#[serde(deny_unknown_fields)]
+pub struct InfluxDBLogsConfig {
+    pub namespace: String,
+    pub endpoint: String,
+    #[serde(default)]
+    pub tags: Vec<String>,
+    #[serde(flatten)]
+    pub influxdb1_settings: Option<InfluxDB1Settings>,
+    #[serde(flatten)]
+    pub influxdb2_settings: Option<InfluxDB2Settings>,
+    #[serde(
+        skip_serializing_if = "crate::serde::skip_serializing_if_default",
+        default
+    )]
+    pub encoding: EncodingConfigWithDefault<Encoding>,
+    #[serde(default)]
+    pub batch: BatchBytesConfig,
+    #[serde(default)]
+    pub request: TowerRequestConfig,
+}
+
+lazy_static! {
+    static ref REQUEST_DEFAULTS: TowerRequestConfig = TowerRequestConfig {
+        retry_attempts: Some(5),
+        ..Default::default()
+    };
+}
+
+#[derive(Deserialize, Serialize, Debug, Eq, PartialEq, Clone, Derivative)]
+#[serde(rename_all = "snake_case")]
+#[derivative(Default)]
+pub enum Encoding {
+    #[derivative(Default)]
+    Default,
+}
+
+inventory::submit! {
+    SinkDescription::new_without_default::<InfluxDBLogsConfig>("influxdb_logs")
+}
+
+#[typetag::serde(name = "influxdb_logs")]
+impl SinkConfig for InfluxDBLogsConfig {
+    fn build(&self, cx: SinkContext) -> crate::Result<(super::RouterSink, super::Healthcheck)> {
+        let healthcheck = healthcheck(
+            self.clone().endpoint,
+            self.clone().influxdb1_settings,
+            self.clone().influxdb2_settings,
+            cx.resolver(),
+        )?;
+
+        let batch = self.batch.unwrap_or(bytesize::mib(1u64), 1);
+        let request = self.request.unwrap_with(&REQUEST_DEFAULTS);
+
+        let sink =
+            BatchedHttpSink::new(self.clone(), Buffer::new(false), request, batch, None, &cx)
+                .sink_map_err(|e| error!("Fatal influxdb_logs sink error: {}", e));
+
+        Ok((Box::new(sink), healthcheck))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn sink_type(&self) -> &'static str {
+        "influxdb_logs"
+    }
+}
+
+impl HttpSink for InfluxDBLogsConfig {
+    type Input = Vec<u8>;
+    type Output = Vec<u8>;
+
+    fn encode_event(&self, event: Event) -> Option<Self::Input> {
+        let mut output = String::new();
+        let mut event = event.into_log();
+
+        let tag_keys: Vec<String> = [
+            vec!["host".to_owned(), "source_type".to_owned()],
+            self.tags.clone(),
+        ]
+        .concat();
+
+        // Measurement
+        let measurement = encode_namespace(&self.namespace, &"vector");
+
+        // Timestamp
+        let timestamp =
+            if let Some(Value::Timestamp(ts)) = event.remove(log_schema().timestamp_key()) {
+                encode_timestamp(Some(ts))
+            } else {
+                encode_timestamp(None)
+            };
```

Comment:
Using `match` makes this simpler:
```rust
let timestamp = encode_timestamp(match event.remove(log_schema().timestamp_key())) {
    Some(Value::Timestamp(ts)) => Some(ts),
    _ => None,
});
```
(It would be even shorter with `map`, but you have to match on the value inside).

Reply:
Done